module/behaviour/skills/HeadBehaviourSoccer/src/HeadBehaviourSoccer.hpp

```diff
@@ -41,113 +41,33 @@ namespace module::behaviour::skills {
-                            const message::input::Image::Lens& lens);
-        void updateHeadPlan(const message::motion::KinematicsModel& kinematicsModel,
-                            const message::vision::Goals& fixationObjects,
-                            const bool& search,
-                            const message::input::Sensors& sensors,
-                            const Eigen::Matrix3d& headToIMUSpace,
-                            const message::input::Image::Lens& lens);
-
-        /*! @brief Converts from camera space direction to IMU space direction
-         */
-        static Eigen::Vector2d getIMUSpaceDirection(const Eigen::Vector2d& screenAngles,
-                                                    const Eigen::Matrix3d& headToIMUSpace);
-
-        /*! @brief Gets points which allow for simultaneous search and viewing of key objects
-         */
-        std::vector<Eigen::Vector2d> getSearchPoints(const message::motion::KinematicsModel& kinematicsModel,
-                                                     const message::vision::Balls& fixationObjects,
-                                                     const message::behaviour::SoccerObjectPriority::SearchType& sType,
-                                                     const message::input::Sensors& sensors,
-                                                     const message::input::Image::Lens& lens);
-        std::vector<Eigen::Vector2d> getSearchPoints(const message::motion::KinematicsModel& kinematicsModel,
-                                                     const message::vision::Goals& fixationObjects,
-                                                     const message::behaviour::SoccerObjectPriority::SearchType& sType,
-                                                     const message::input::Sensors& sensors,
-                                                     const message::input::Image::Lens& lens);
-
-        /*! @brief Combines a collection of vision objects. The screen resulting screen angular region is the
-         * bounding box of the objects
-         */
-        message::vision::Ball combineVisionObjects(const message::vision::Balls& obs);
-        message::vision::Goal combineVisionObjects(const message::vision::Goals& obs);
-
-        /*! @brief Gets a bounding box in screen angular space of a set of vision objects
-         */
-        static utility::math::geometry::Quad<double, 2, 1> getScreenAngularBoundingBox(
-            const message::vision::Balls& obs);
-        static utility::math::geometry::Quad<double, 2, 1> getScreenAngularBoundingBox(
-            const message::vision::Goals& obs);
-
-        bool orientationHasChanged(const message::input::Sensors& sensors);
-
-
         // CONFIG - loaded elsewhere
-        float max_yaw   = 0.0f;
-        float min_yaw   = 0.0f;
-        float max_pitch = 0.0f;
-        float min_pitch = 0.0f;
+        float max_yaw           = 0.0f;
```

Comment:
If these are loaded from the config what is the purpose of giving them values here? Is this in case the config does not exist?

Reply:
Always initialise your variables! Clang tidy will probably complain.

Reply:
Yes mum 😅